engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/systemchannels/SensitiveContentChannel.java

```diff
@@ -0,0 +1,173 @@
+  private int deserializeContentSensitivity(String contentSensitivityName) {
+    switch (contentSensitivityName) {
+      case AUTO_SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_AUTO;
+      case SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_NOT_SENSITIVE;
+      case NOT_SENSITIVE_CONTENT_SENSITIVITY:
+        return View.CONTENT_SENSITIVITY_SENSITIVE;
+      default:
+        throw new IllegalArgumentException(
+            "contentSensitivityName "
+                + contentSensitivityName
+                + " not known to the SensitiveContentChannel.");
+    }
+  }
+
+  // The annotation to suppress the "InlinedApi" lint prevents lint warnings
+  // caused by usage of Android 35 APIs. This call is safe because these values
+  // will never be used to set content sensitivity for a View via a
+  // SensitiveContentMethodHandler because it must ensure a device is API 35+ before
+  // doing so to avoid exceptions on other devices.
+  @SuppressLint({"InlinedApi"})
+  private String serializeContentSensitivity(int contentSensitivityValue) {
+    switch (contentSensitivityValue) {
+      case View.CONTENT_SENSITIVITY_AUTO:
+        return AUTO_SENSITIVE_CONTENT_SENSITIVITY;
+      case View.CONTENT_SENSITIVITY_NOT_SENSITIVE:
+        return NOT_SENSITIVE_CONTENT_SENSITIVITY;
+      case View.CONTENT_SENSITIVITY_SENSITIVE:
+        return SENSITIVE_CONTENT_SENSITIVITY;
+      default:
+        throw new IllegalArgumentException(
+            "Android View content sensitivity constant with value "
+                + contentSensitivityValue
+                + " not known to the SensitiveContentChannel.");
+    }
+  }
+
+  public SensitiveContentChannel(@NonNull DartExecutor dartExecutor) {
+    channel =
+        new MethodChannel(dartExecutor, "flutter/sensitivecontent", StandardMethodCodec.INSTANCE);
+    channel.setMethodCallHandler(parsingMethodHandler);
+  }
+
+  /**
+   * Sets the {@link SensitiveContentMethodHandler} which receives all requests to get and set a
+   * particular content sensitivty level sent through this channel.
+   */
+  public void setSensitiveContentMethodHandler(
+      @Nullable SensitiveContentMethodHandler sensitiveContentMethodHandler) {
+    this.sensitiveContentMethodHandler = sensitiveContentMethodHandler;
+  }
+
+  public interface SensitiveContentMethodHandler {
+    /**
+     * Requests that a native Flutter Android {@code View} sets its content sensitivity level to
+     * {@code requestedContentSensitivity}.
+     */
+    void setContentSensitivity(@NonNull int requestedContentSensitivity);
+
+    /** Returns the current content sensitivity level of a Flutter Android {@code View}. */
+    int getContentSensitivity();
+
+    /**
+     * Returns whether or not marking content sensitivity is supported on the device.
+     *
+     * <p>This value is static and will not change while a Flutter app runs.
+     */
+    boolean isSupported();
```

Comment:
if this is about marking (as opposed to getting) then maybe it should be named `isSetContentSensitvitySupported` 
 
 I realize this might change a lot of other method names. What do you think?

Reply:
Oops! It is for getting/setting! Updated the docs.